### PR TITLE
feat: support redis monitor

### LINKED ISSUE #1376: Is mongo db / redis supported?
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

just read title

### 🐻 Uptime-Kuma Version

1.10.0

### 💻 Operating System and Arch

irrelevant for my question

### 🌐 Browser

irrelevant for my question

### 🐋 Docker Version

irrelevant for my question

### 🟩 NodeJS Version

irrelevant for my question

### PR DESCRIPTION
**Pull request by long2ice**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #1376 for redis

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Translation update
- Other
- This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

File: server/model/monitor.js [hunk 1]
```diff
@@ -120,6 +122,7 @@ class Monitor extends BeanModel {
                 basic_auth_pass: this.basic_auth_pass,
                 pushToken: this.pushToken,
                 databaseConnectionString: this.databaseConnectionString,
+                redisConnectionString: this.redisConnectionString,
```
Comment (louislam) [inline server/model/monitor.js:125]: Please reuse `databaseConnectionString`

**Comment by louislam** [comment]
Uptime Kuma crashed if the target is not a Redis server.

Example: `redis://uptime.kuma.pet:443`

```
node:events:491
      throw er; // Unhandled 'error' event
      ^

SocketClosedUnexpectedlyError: Socket closed unexpectedly
    at Socket.<anonymous> (C:\Users\louis\Dropbox\My Projects\uptime-kuma\node_modules\@redis\client\dist\lib\client\socket.js:181:118)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:513:28)
    at TCP.<anonymous> (node:net:298:12)
Emitted 'error' event on Commander instance at:
    at RedisSocket.<anonymous> (C:\Users\louis\Dropbox\My Projects\uptime-kuma\node_modules\@redis\client\dist\lib\client\index.js:360:14)
    at RedisSocket.emit (node:events:513:28)
    at RedisSocket._RedisSocket_onSocketError (C:\Users\louis\Dropbox\My Projects\uptime-kuma\node_modules\@redis\client\dist\lib\client\socket.js:204:10)
    at Socket.<anonymous> (C:\Users\louis\Dropbox\My Projects\uptime-kuma\node_modules\@redis\client\dist\lib\client\socket.js:181:107)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:513:28)
    at TCP.<anonymous> (node:net:298:12)

```

**Comment by louislam** [comment]
> I test postgres with same operation, which also crashed

Thanks for testing, both may need to fix.

But that is weird, because it should be inside try...catch.

**Comment by louislam** [comment]
Just have a quick look, I think the library is not actually a fully await/async library, you still need to add the event handler. 
Because of this, I would recommend that you should change the `redisPingAsync` function to return promise instead of async function. So that it can `reject(error)` in the event handler.

Also there are 4 more possible events, better add them too for safety.

image


image

**Comment by louislam** [comment]
Remark for testing:

> docker run --rm -p 6379:6379 redis

File: server/util-server.js [hunk 2]
```diff
@@ -353,6 +354,29 @@ exports.radius = function (
         ],
     });
 };
+/**
```
Comment (Computroniks) [inline server/util-server.js:357]:
```suggestion

/**
```
Preserve line between function and previous statement.

**Review by Computroniks — CHANGES_REQUESTED** [review]

**Pull request merged**